jx_sqlite/expressions.py

```diff
@@ -144,7 +144,14 @@ def to_sql(self, schema, not_null=False, boolean=False):
                 if r.sql[t] == None:
                     acc.append("(" + l.sql[t] + ") IS NULL")
                 else:
-                    acc.append("((" + l.sql[t] + ") = (" + r.sql[t] + ") OR ((" + l.sql[t] + ") IS NULL AND (" + r.sql[t] + ") IS NULL))")
+                    if ".$" in l.sql[t] and ".$" in r.sql[t]:
```

Comment:
This is a strange check; it will not work. You have a high chance of detecting false positives given the complexity of the SQL that is possible.